clang/include/clang/Basic/AtomicOptions.def

```diff
@@ -0,0 +1,19 @@
+//===--- AtomicOptions.def - Atomic Options database -------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This file defines the Atomic language options. Users of this file
+// must define the OPTION macro to make use of this information.
+#ifndef OPTION
+#  error Define the OPTION macro to handle atomic language options
+#endif
+
+// OPTION(name, type, width, previousName)
+OPTION(NoRemoteMemory, bool, 1, First)
+OPTION(NoFineGrainedMemory, bool, 1, NoRemoteMemory)
+OPTION(IgnoreDenormalMode, bool, 1, NoFineGrainedMemory)
```

Comment:
Can't say I'm a fan of negative options.
IIUIC, from the comments on the RFC (https://discourse.llvm.org/t/rfc-add-clang-atomic-control-options-and-pragmas/80641/23), one of the reasons for sticking with them is that we currently have no way to have those options enabled by default.

Would it make sense to improve this ability to specify the default value, and then stick to positive assertions and their negations in the control knobs' names? This sounds like something that would be generally useful improvement applicable beyond this patch, and it should address your concern that dropping those hints should preserve safe conservative behavior.

Reply:
I understand the desire to avoid negative naming in APIs. However, in this case, the negative naming (NoRemoteMemory, NoFineGrainedMemory) actually better reflects the IR structure. By default, atomic instructions in IR have no metadata, which corresponds to the conservative worst case (memory may be remote/fine-grained). The metadata we add serves as explicit performance hints promising the memory is not remote/fine-grained.
If we used positive names like RemoteMemory, it would imply there's a default remote_memory metadata in IR, which isn't true and could cause confusion. The current negative naming makes it clearer that we're adding explicit metadata to opt out of the conservative default behavior.

Reply:
OK.